src/utils/ddb/ddb_vos.c

```diff
@@ -1771,29 +1771,148 @@ sync_cb(struct ddbs_sync_info *info, void *cb_args)
 		 */
 		blob_size = info->dsi_cluster_nr * info->dsi_cluster_size;
 	} else {
-		blob_size = pool_info->spi_blob_sz[st];
-		smd_pool_free_info(pool_info);
+		blob_size = pinfo->spi_blob_sz[st];
+		if (pool_info == NULL)
+			smd_pool_free_info(pinfo);
 	}
 
 	/* Try to delete the target first */
-	rc = smd_pool_del_tgt(pool_id, info->dsi_hdr->bbh_vos_id, st);
+	rc = smd_pool_del_tgt(pool_id, tgt_id, st);
 	if (!SUCCESS(rc))
 		/* Ignore error for now ... might not exist*/
 		D_WARN("delete target failed: " DF_RC "\n", DP_RC(rc));
 
-	rc = smd_pool_add_tgt(pool_id, info->dsi_hdr->bbh_vos_id, info->dsi_hdr->bbh_blob_id, st,
+	rc = smd_pool_add_tgt(pool_id, tgt_id, blob_id, st,
 			      blob_size, 0, false);
 	if (!SUCCESS(rc)) {
 		D_ERROR("add target failed: "DF_RC"\n", DP_RC(rc));
 		args->sync_rc = rc;
 		return;
 	}
 
+	if (st == SMD_DEV_TYPE_META) {
+		rc = smd_pool_mark_ready(pool_id);
+		if (!SUCCESS(rc)) {
+			D_ERROR("mark target ready failed: "DF_RC"\n", DP_RC(rc));
+			args->sync_rc = rc;
+			return;
+		} 
+	}
+
 	if (args->sync_complete_cb) {
 		rc = args->sync_complete_cb(args->sync_cb_args, pool_id,
-					    info->dsi_hdr->bbh_vos_id,
-					    info->dsi_hdr->bbh_blob_id,
-					    blob_size, info->dsi_dev_id);
+					    tgt_id, blob_id, blob_size, info->dsi_dev_id, st);
+	}
+}
+
+static void
+sync_cb_data(struct ddbs_sync_info *info, void *cb_args)
+{
+	do_sync_cb(info->dsi_hdr->bbh_pool, info->dsi_hdr->bbh_vos_id, 
+				 	info->dsi_hdr->bbh_blob_id, info->st, NULL, info, cb_args);
+}
+
+static struct smd_pool_info *
+find_pool_by_blobid_v1(struct meta_header_v1 *hdr, d_list_t pool_list, int pool_list_cnt)
+{
+	struct smd_pool_info    *pool_info = NULL;
+	struct smd_pool_info    *tmp = NULL;
+	int i = 0;
+	int found = 0;
+	int cnt = 0;
+
+	enum smd_dev_type st;
+	uint32_t target_id = hdr->mh_vos_id;
+	uint64_t blob_id[SMD_DEV_TYPE_MAX];
+	blob_id[SMD_DEV_TYPE_DATA] = hdr->mh_data_blobid;
+	blob_id[SMD_DEV_TYPE_META] = hdr->mh_meta_blobid;
+	blob_id[SMD_DEV_TYPE_WAL] = hdr->mh_wal_blobid;
+
+	d_list_for_each_entry_safe(pool_info, tmp, &(pool_list), spi_link) {
+		for (st = SMD_DEV_TYPE_DATA; st < SMD_DEV_TYPE_MAX; st++) {
+			for (i = 0; i < pool_info->spi_tgt_cnt[st]; i++) {
+				if (pool_info->spi_tgts[st][i] == target_id)
+					break;
+			}
+			if (pool_info->spi_blobs[st][i] != blob_id[st])
+				break;
+			found++;
+		}
+
+		// Found 
+		if (found == SMD_DEV_TYPE_MAX) 
+			return pool_info;
+		if (++cnt >= pool_list_cnt)
+			break;
+	}
+
+	return NULL;
+}
+
+static void
+sync_cb_meta(struct ddbs_sync_info *info, void *cb_args)
+{
+	struct dv_sync_cb_args	*args = cb_args;
+	struct smd_pool_info    *pool_info = NULL;
+	struct meta_header * hdr = info->dsi_meta_hdr;
+	uuid_t pool_id;
+
+    if (hdr->mh_version == 1) {
+		/* For Phase 1, No pool id in meta blob. get from Data Blob */
```

Comment:
I'm wondering if we should just simply return error (DER_NOTSUPPORTED) for such case? Otherwise, it could be complicated since you have to ensure data SSD/blobs are scanned before meta SSD/blobs, what do you think?